utils.py

```diff
@@ -157,27 +159,25 @@ def time(cmd):
 def weather(cmd):
 	if not cmd.args:
 		return
-	split = cmd.args.split()
-	if split[0].startswith('-'):
-		flags = split[0][1:]
-		location = ' '.join(split[1:])
-	elif split[-1].startswith('-'):
-		flags = split[-1][1:]
-		location = ' '.join(split[:-1])
-	else:
-		flags = '1Fp'
-		location = cmd.args
-		if location.isdecimal() and len(location) == 5:
-			location += '-us'
-	url = 'https://wttr.in/%s.png?%s' % (urllib.parse.quote_plus(location), flags)
+	flags = 'format=**%l:**+%c+++🌡+`%t(%f)`++💦+`%h`++💨+`%w`++**☔**+`%p/3h`++**UVI:**+`%u`\
+				\n**Time:**+`%T`++**Sunrise:**+`%S`++**Sunset:**+`%s`++**Moon:**+%m'
+	location = cmd.args
+	if location.isdecimal() and len(location) == 5:
+		location += '-us'
+	url = f'https://wttr.in/{urllib.parse.quote_plus(location)}?{flags}'
 	try:
 		response = rs.get(url)
 		response.raise_for_status()
 	except Exception:
-		cmd.reply('%s: error getting weather at %s' % (cmd.sender['pretty_name'], url),
-				{'description': '```%s```' % traceback.format_exc()[-500:]})
+		if response.status_code == 503:
+			cmd.reply(f'{cmd.sender["pretty_name"]}: service unavailable for {location}')
+		elif response.status_code == 404:
```

Comment:
this feels kinda questionable... but it does save repeating `return`s. the alternative is
```py
try:
	response = rs.get(url)
	if response.status_code == 503:
		cmd.reply(f'{cmd.sender["pretty_name"]}: service unavailable for {location}')
		return
	elif response.status_code == 404:
		cmd.reply(f'{cmd.sender["pretty_name"]}: {location} not found')
		return
    response.raise_for_status()
except Exception:
	cmd.reply(f'{cmd.sender["pretty_name"]}: error getting weather at {url}',
			{'description': f'```{traceback.format_exc()[-500:]}```'})
    return
```
I leave it up to you to decide which is better
(or, you could introduce some kinda `BotCommandException` and change how the bot handles exceptions... would clean up the control flow a bit)

Reply:
Yours is better, I think, it avoids raising exceptions for things that are actually handled properly.